### ISSUE TITLE
Add Bootstrapped Connections to Persistent Peers

### ISSUE THREAD

**Issue by cfromknecht** [lightningnetwork/lnd#451]
Overview
=======

At the moment, when a peer disconnects from lnd, it will be handled in one of two ways depending on whether the connection is persistent or non-persistent. A _persistent connection_ is one that our daemon will try and reestablish if a disconnect occurs, while a non-persistent one will be allowed to disconnect without any further action. The goal of this issue is to modify the server to add successfully connected peers from the DNS bootstrapper to the set of persistent peers. This will allow lnd to maintain better network connectivity—even if it has no other persistent connections were opened via the process of channel creation—and its bootstrapped peers get disconnected intermittently.

Steps To Completion
================

- [ ]  Re-establishment of a persistent connection occurs in the server's `peerTerminationWatcher`, predicated on whether or not the peer's pubkey string is present in the server's `persistentPeers` map. There are currently two instances inside the `peerBootstrapper` where a go routine is spawned to `Dial` a peer, and then execute `OutboundPeerConnected` upon success. In order to make these connections persistent, the peer's pubkey string should be added to the `persistentPeers` map before executing the callback. This will cause the connection to be reestablished by way of the connection manager in the event that the `peerTerminationWatcher` exits.

**Comment by cfromknecht** [comment]
@bajohns has expressed interest in tackling this one, let me know if anything needs clarification!

**Comment by bajohns** [comment]
@cfromknecht I will review this and start working through setting up persistent peers.

It seems as though the assumption is that peers will stay online or at least that 1/3 of the bootstrapped peers will. In testnet that doesn't seem to hold as peers not only drop but a set of nodes can become isolated from the entire network. This isolation occurs if both nodes lose all peers but each other. Before your explanation above, I considered rerunning the bootstrap process if peer count hit zero or if messages frequency dropped below a certain threshold. 

By example, now I am running a node that is listening on a public IP. It successfully bootstraps three peers and they disconnect over time. Over the last twenty four hours I noticed the peer count didn't go to zero but the message volume did. Upon process restart I experienced a flood of over twelve hours worth of messages.

I will start with persistent peers above and let you know my results.
